### PR TITLE
feat(nx): Make nx preset standalone

### LINKED ISSUE #24: Make preset generator standalone
The core present is so simple, and we've modified it so much, we should consider just making our generator independent

### PR DESCRIPTION
**Pull request by luxaritas**
Resolves #24 

Of note:
* Added a new `bootstrap` package to use instead of `create-nx-workspace`. I opted to do this for a few reasons, including allowing us more control over the nx version being used, avoiding surfacing options we wind up not supporting/overriding, and the ability to show generator prompts in our preset.
* Fleshed out the preset generator to handle all the things we relied on the nx core generator to include (plus some tweaks as I was working through them, eg adjusting the default gitignore)
* Removed the preset-eterna generator in favor of an option to set defaults within the normal preset generator
* Made nx a dependency of @eternagame/nx, so that our plugin completely handles the nx version being used
* Fixed syntax typo in ts-iso generator (`export`, not `exports`)
* `vite` package fixes for bugs noticed in developing `bootstrap`, where dependencies weren't being consistently externalized for node apps and shebangs weren't able to be handled

### CODE REVIEW

**Comment by luxaritas** [comment]
A couple other issues I noticed as I was testing that I've resolved:
* We don't want to have the `@nrwl/workspace` plugin used in this workspace any more, as this workspace should be compliant with the new defaults
* nx pinned some dependencies, but we should be fine treating them as `^` instead
* When upgrading nx, an import in nx-spawn were incorrectly updated
* `build-watch` is indeed needed for `app-node`, as the build and serve steps are run separately, unlike `app-web`

**Review by tkaragianes — COMMENTED** [review]
Just a couple simple questions I had.

File: packages/nx/src/generators/preset/files/.vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "recommendations": [
+    "nrwl.angular-console",
```
[the remark this answers is not in the document]
Comment (luxaritas) [reply packages/nx/src/generators/preset/files/.vscode/extensions.json:3]: The extension is misnamed - this is the nx console, which provides a UI for nx commands

File: packages/nx/src/generators/ts-iso/files/src/index.ts__tmpl__ [hunk 2]
```diff
@@ -1,3 +1,3 @@
 console.log("Hello world");
```
[the remark this answers is not in the document]
Comment (luxaritas) [reply packages/nx/src/generators/ts-iso/files/src/index.ts__tmpl__:1]: No, this is just what I decided to use as the default code, that way it does _something_ - though in retrospect, an `iso` package probably can't be run, so maybe it should just export something? `export const x = 'Hello world'`?

**Comment by luxaritas** [comment]
So I wound up opting to include nx as a dependency in the generated package.json since it really should be an (albeit pinned) peer dep in `@eternagame/nx` - we do directly use the CLI within our app. However, I neglected to add it back to the package.json of this repo - need to fix that

**Comment by luxaritas** [comment]
Never mind, it is there >.<

**Pull request merged**